connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -132,12 +132,31 @@ class SparkConnectPlanner(session: SparkSession) {
    * wrap such fields into proto messages.
    */
   private def transformSample(rel: proto.Sample): LogicalPlan = {
+    val plan = if (rel.getInput.hasRandomSplit) {
```

Comment:
I'm a bit confused, This function is matching `Sample` but checks `RandomSplit`. Aren't they the different proto messages?

Reply:
Because we should construct the child plan of `Sample`. The child plan can be easily built in `SparkConnectPlanner` but hard in connect's `dataframe`.